### PR TITLE
Fixing RenderObject to reflect name changes done in CustomForwardRenderer asset

### PR DESCRIPTION
**Pull request by lukaschod**
### Purpose of this PR
Fix for case https://fogbugz.unity3d.com/f/cases/1246256/.

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: No need for automated test as UI fix.

**Yamato**: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252Frenderobjects-rename

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/ScriptableRendererDataEditor.cs [hunk 1]
```diff
@@ -118,7 +118,17 @@ private void DrawRendererFeature(int index, ref SerializedProperty prop)
...
                     if (EditorGUI.EndChangeCheck())
+                    {
                         m_SaveAsset = true;
+
+                        // We need to update sub-asset name
+                        obj.name = propertyName.stringValue;
+                        AssetDatabase.SaveAssets();
+
+                        // Triggers update for sub-asset name change
+                        // TODO: Investigate for better alternative as AssetDatabase for some reason do not trigger this change
```
Comment (martint-unity) [inline com.unity.render-pipelines.universal/Editor/ScriptableRendererDataEditor.cs:129]: We shouldn't have TODO's in our code if we do not really really really have to.

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 2]
```diff
@@ -206,6 +206,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed an issue where camera stacking with MSAA on OpenGL resulted in a black screen. [case 1250602](https://issuetracker.unity3d.com/issues/urp-camera-stacking-results-in-black-screen-when-msaa-and-opengl-graphics-api-are-used)
 - Optimized shader compilation times by compiling different variant sets for vertex and fragment shaders.
 - Fixed shadows for additional lights by limiting MAX_VISIBLE_LIGHTS to 16 for OpenGL ES 2.0 and 3.0 on mobile platforms. [case 1244391](https://issuetracker.unity3d.com/issues/android-urp-spotlight-shadows-are-not-being-rendered-on-adreno-330-and-320-when-built)
+- Fixing RenderObject to reflect name changes done in CustomForwardRenderer asset. [case 1246256]
```
Comment (martint-unity) [inline com.unity.render-pipelines.universal/CHANGELOG.md:209]: This should also have the public issuetracker

Comment (martint-unity) [reply com.unity.render-pipelines.universal/CHANGELOG.md:209]: This needs to be fixed before merge

**Comment by martint-unity** [comment]
Sorry, looks like I doubled up on some feedback above :)

**Comment by lukaschod** [comment]
Updated the changelog with issuetracker, going to do same with other PRs. Thanks

File: com.unity.render-pipelines.universal/Editor/ScriptableRendererDataEditor.cs (hunk 1 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Editor/ScriptableRendererDataEditor.cs:129]: Well this one is tricky, sub asset renaming think could be a bug. However I could not get clarifying response from asset team. Going to try poking them again.

Comment (lukaschod) [reply com.unity.render-pipelines.universal/Editor/ScriptableRendererDataEditor.cs]: I created favro card for this and https://fogbugz.unity3d.com/f/cases/1260165/.

**Comment by lukaschod** [comment]
You can continue the review.

**Comment by lukaschod** [comment]
@martin-unity3d could u recheck is it okay now

**Comment by phi-lira** [comment]
@martint-unity can you review this so we move on?

**Pull request merged**